substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1990,6 +1983,47 @@ pub mod pallet {
 			}
 			Ok(Some(T::WeightInfo::deprecate_controller_batch(controllers.len() as u32)).into())
 		}
+
+		/// TODO: docs
+		/// Basically another call for `bond`, but it's implied that the funds are already locked
+		/// and `pallet-staking` doesn't need to lock them here.
+		#[pallet::call_index(29)]
+		#[pallet::weight(T::WeightInfo::bond())]
+		pub fn delegated_bond(
+			origin: OriginFor<T>,
+			#[pallet::compact] value: BalanceOf<T>,
+			payee: RewardDestination<T::AccountId>,
+		) -> DispatchResult {
+			// To make this truely decoupled, we would need a `EnsureDelegatorOrigin` type on
```

Comment:
How would you do `EnsureDelegatorOrigin`? I am guessing we will use a signed extension that calls into Delegator Staking pallet? 
I don't see how it is better than decoupling it via a glue trait. And the worse thing about this is `SignedExtension` should be as computationally cheap as possible since it gets called for every dispatchable and doing a database read on it is a significant overhead.